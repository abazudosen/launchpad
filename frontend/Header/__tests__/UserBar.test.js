/* @flow */

/**
 * @jest-environment node
 */

import React from 'react';
import UserBar from '../UserBar';
import { mount } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

expect.addSnapshotSerializer(serializer);

describe('User Bar', () => {
  test('renders normal', () => {
    const component = mount(
      <UserBar
        user={{
          id: 'test-id',
          githubUsername: 'example',
          pads: [],
        }}
        onLogin={() => {}}
        onLogout={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();

    component.find('HeaderButton').simulate('click');
    expect(component).toMatchSnapshot();

    component.find('HeaderButton').simulate('click');
    expect(component).toMatchSnapshot();
  });
});
