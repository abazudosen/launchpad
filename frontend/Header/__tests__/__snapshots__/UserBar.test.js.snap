// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Bar renders normal 1`] = `
<UserBar
  onLogin={[Function]}
  onLogout={[Function]}
  user={
    Object {
      "githubUsername": "example",
      "id": "test-id",
      "pads": Array [],
    }
  }
>
  <div
    className="UserBar"
  >
    <HeaderDropdown
      button={
        <HeaderButton
          active={false}
          onClick={[Function]}
        >
          <span
            className="icon-user"
          />
          <span>
            example
          </span>
          <span
            className="UserBar-Arrow"
          >
            ▾
          </span>
        </HeaderButton>
      }
      onClose={[Function]}
      open={false}
    >
      <div
        className="HeaderDropdown"
      >
        <HeaderButton
          active={false}
          onClick={[Function]}
        >
          <button
            className="HeaderButton"
            onClick={[Function]}
          >
            <div
              className="HeaderButton-Inner"
            >
              <span
                className="icon-user"
              />
              <span>
                example
              </span>
              <span
                className="UserBar-Arrow"
              >
                ▾
              </span>
            </div>
          </button>
        </HeaderButton>
        <div
          className="HeaderDropdown-Dropdown"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <HeaderDropdownButton
            onClick={[Function]}
          >
            <button
              className="HeaderDropdownButton"
              onClick={[Function]}
            >
              Sign out
            </button>
          </HeaderDropdownButton>
          <HeaderDropdownButton>
            <button
              className="HeaderDropdownButton"
            >
              <a
                href="/list"
              >
                Pads you created
              </a>
            </button>
          </HeaderDropdownButton>
        </div>
      </div>
    </HeaderDropdown>
  </div>
</UserBar>
`;

exports[`User Bar renders normal 2`] = `
<UserBar
  onLogin={[Function]}
  onLogout={[Function]}
  user={
    Object {
      "githubUsername": "example",
      "id": "test-id",
      "pads": Array [],
    }
  }
>
  <div
    className="UserBar"
  >
    <HeaderDropdown
      button={
        <HeaderButton
          active={true}
          onClick={[Function]}
        >
          <span
            className="icon-user"
          />
          <span>
            example
          </span>
          <span
            className="UserBar-Arrow"
          >
            ▾
          </span>
        </HeaderButton>
      }
      onClose={[Function]}
      open={true}
    >
      <div
        className="HeaderDropdown"
      >
        <HeaderButton
          active={true}
          onClick={[Function]}
        >
          <button
            className="HeaderButton HeaderButton--active"
            onClick={[Function]}
          >
            <div
              className="HeaderButton-Inner"
            >
              <span
                className="icon-user"
              />
              <span>
                example
              </span>
              <span
                className="UserBar-Arrow"
              >
                ▾
              </span>
            </div>
          </button>
        </HeaderButton>
        <div
          className="HeaderDropdown-Dropdown"
          style={Object {}}
        >
          <HeaderDropdownButton
            onClick={[Function]}
          >
            <button
              className="HeaderDropdownButton"
              onClick={[Function]}
            >
              Sign out
            </button>
          </HeaderDropdownButton>
          <HeaderDropdownButton>
            <button
              className="HeaderDropdownButton"
            >
              <a
                href="/list"
              >
                Pads you created
              </a>
            </button>
          </HeaderDropdownButton>
        </div>
      </div>
    </HeaderDropdown>
  </div>
</UserBar>
`;

exports[`User Bar renders normal 3`] = `
<UserBar
  onLogin={[Function]}
  onLogout={[Function]}
  user={
    Object {
      "githubUsername": "example",
      "id": "test-id",
      "pads": Array [],
    }
  }
>
  <div
    className="UserBar"
  >
    <HeaderDropdown
      button={
        <HeaderButton
          active={false}
          onClick={[Function]}
        >
          <span
            className="icon-user"
          />
          <span>
            example
          </span>
          <span
            className="UserBar-Arrow"
          >
            ▾
          </span>
        </HeaderButton>
      }
      onClose={[Function]}
      open={false}
    >
      <div
        className="HeaderDropdown"
      >
        <HeaderButton
          active={false}
          onClick={[Function]}
        >
          <button
            className="HeaderButton"
            onClick={[Function]}
          >
            <div
              className="HeaderButton-Inner"
            >
              <span
                className="icon-user"
              />
              <span>
                example
              </span>
              <span
                className="UserBar-Arrow"
              >
                ▾
              </span>
            </div>
          </button>
        </HeaderButton>
        <div
          className="HeaderDropdown-Dropdown"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <HeaderDropdownButton
            onClick={[Function]}
          >
            <button
              className="HeaderDropdownButton"
              onClick={[Function]}
            >
              Sign out
            </button>
          </HeaderDropdownButton>
          <HeaderDropdownButton>
            <button
              className="HeaderDropdownButton"
            >
              <a
                href="/list"
              >
                Pads you created
              </a>
            </button>
          </HeaderDropdownButton>
        </div>
      </div>
    </HeaderDropdown>
  </div>
</UserBar>
`;
